src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/AbstractToMethodConverter.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.ConvertLinq.ConvertForEachToLinqQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery
+{
+    internal abstract class AbstractToMethodConverter : AbstractConverter
+    {
+        // It is "item" for for "list.Add(item)"
+        // It can be anything for "counter++". It will be ingored in the case.
+        private readonly ExpressionSyntax _selectExpression;
+
+        // It is "list" for "list.Add(item)"
+        // It is "counter" for "counter++"
+        private readonly ExpressionSyntax _modifyingExpression;
+
+        // Trivia around "counter++;" or "list.Add(item);". Required to keep comments.
+        private readonly SyntaxTrivia[] _trivia;
```

Comment:
interestin that is' "around", but is just a simple list.  how do you know what is preceding vs trailing trivia?  or do you just not care?

Reply:
On the current stage, I do not care about separating leading and trailing trivia for the expression because the expression is totally gone. We add those comments together after the query expression generated. If we get an example we need to support, we can make an improvement here.

---
In reply to: [195844271](https://github.com/dotnet/roslyn/pull/27130#discussion_r195844271) [](ancestors = 195844271)